mungegithub/mungers/dibs-nack-handler.go

```diff
@@ -0,0 +1,195 @@
+			//check if they are a valid reviewer if so, assign the user. if not, explain why
+			if _, ok := potential_owners[comment.User.String()]; ok {
+				glog.Infof("Assigning %v to review PR#%v", *comment.User.Login, obj.Issue.Number)
+				obj.AssignPR(comment.User.String())
+				return
+			} else {
+				// add comment explaining to them they can't be reviewer
+				glog.Infof("%v requested to be added as a reviewer but is not a potential reviewer")
+			}
+		}
+
+		if isNackComment(fields) {
+			//check if they are already an assigned reviewer. if so, remove them.  if not, do nothing.
+			glog.Infof("Removing %v as an reviewer for PR#%v", *comment.User.Login, obj.Issue.Number)
+			for _, assignee := range obj.Issue.Assignees {
+				//remove the assignee
+				if comment.User.ID == assignee.ID{
+					is := goGithub.IssuesService{}
+					toDelete := []string{*comment.User.Name}
+					is.RemoveAssignees(OWNER, *obj.Issue.Repository.Name, *obj.Issue.Number, toDelete)
+				}
+			}
+			continue
+		}
+
+		// check if someone manually removed the lgtm label after the `/lgtm` comment
+		// and honor it.
+		if removeLGTMTime != nil && removeLGTMTime.After(*comment.CreatedAt) {
+			return
+		}
+
+		// TODO: support more complex policies for multiple reviewers.
+		// See https://github.com/kubernetes/contrib/issues/1389#issuecomment-235161164
+		obj.AddLabel(lgtmLabel)
+		return
+	}
+}
+
+
+func (h *DibsNackHandler) removeIfNack(obj *github.MungeObject, comments []*goGithub.IssueComment, events []*goGithub.IssueEvent, reviewers mungerutil.UserSet) {
+	// Get the last time when the someone applied lgtm manually.
+	removeLGTMTime := e.LastEvent(events, e.And{e.RemoveLabel{}, e.LabelName(lgtmLabel), e.HumanActor()}, nil)
+
+	// Assumption: The comments should be sorted (by default from github api) from oldest to latest
+	for i := len(comments) - 1; i >= 0; i-- {
+		comment := comments[i]
+		if !mungerutil.IsValidUser(comment.User) {
+			continue
+		}
+
+		// TODO: An approver should be acceptable.
+		// See https://github.com/kubernetes/contrib/pull/1428#discussion_r72563935
+		if !mungerutil.IsMungeBot(comment.User) && !isReviewer(comment.User, reviewers) {
+			continue
+		}
+
+		fields := getFields(*comment.Body)
+		if isCancelComment(fields) {
+			// "/lgtm cancel" if commented more recently than "/lgtm"
+			return
+		}
+
+		if !isLGTMComment(fields) {
+			continue
+		}
+
+		// check if someone manually removed the lgtm label after the `/lgtm` comment
+		// and honor it.
+		if removeLGTMTime != nil && removeLGTMTime.After(*comment.CreatedAt) {
+			return
+		}
+
+		// TODO: support more complex policies for multiple reviewers.
+		// See https://github.com/kubernetes/contrib/issues/1389#issuecomment-235161164
+		glog.Infof("Adding lgtm label. Reviewer (%s) LGTM", *comment.User.Login)
+		obj.AddLabel(lgtmLabel)
+		return
+	}
+}
+func isDibsComment(fields []string) bool {
+	// Note: later we'd probably move all the bot-command parsing code to its own package.
```

Comment:
pushing back.  This is the way it's done in lgtm.  I'm going to try and be consistent